### PR TITLE
[runtime] Add a fastpath to mono_object_isinst_icall ().

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -276,6 +276,21 @@ mono_marshal_init_tls (void)
...
 {
+	if (!klass)
+		return NULL;
+
+	/* This is called from stelemref so it is expected to succeed */
+	/* Fastpath */
+	if (mono_class_is_interface (klass)) {
+		MonoVTable *vt = obj->vtable;
+
+		if (!klass->inited)
```
Comment (evincarofautumn) [inline mono/metadata/marshal.c]: This is needed to init `klass->interface_id`? When can this happen?

Comment (vargaz) [inline mono/metadata/marshal.c]: Its extracted from mono_object_handle_isinst ().

Comment (kumpera) [inline mono/metadata/marshal.c]: Aren't we better off adding another stelemref wrapper for interfaces?

Comment (vargaz) [inline mono/metadata/marshal.c]: This is called from the 'complex' stelemref wrapper, and its just a fastpath, if the check fails, we still need to do all the other checks.

**Pull request merged**